src/handlers.cpp

```diff
@@ -1192,6 +1192,11 @@ void ImpuRegDataTask::on_get_reg_data_success(CassandraStore::Operation* op)
   // Split the processing depending on whether an HSS is configured.
   if (_cfg->hss_configured)
   {
+    // We need the record to last twice the HSS Re-registration
+    // time, or the max expiry of the registration, which ever one
+    // is longer.
+    int record_ttl = std::max(2 * _cfg->hss_reregistration_time, _cfg->reg_max_expires + 10);
```

Comment:
Should you calculate this once in main.cpp and put it into `_cfg->record_ttl`, instead of doing the calculation in a couple of separate functions?